src/services/auth.js

```diff
@@ -97,15 +116,42 @@ class Auth {
     })
   }
 
-  saveRedirectURL (url) {
-    sessionStorage.setItem('after_login_redirect_url', url)
+  /* State getters and setters */
+  isAuthenticated () {
+    return (
+      this.session &&
+      this.session.expiresAt > new Date().getTime() &&
+      localStorage.userId
+    )
   }
 
-  retrieveRedirectURL () {
-    const url = sessionStorage.getItem('after_login_redirect_url')
+  wasAuthenticated () {
+    return (
+      this.session &&
+      this.session.expiresAt < new Date().getTime() &&
+      localStorage.userId
+    )
+  }
+
+  getAfterLoginRedirectUrl () {
```

Comment:
It seems a bit misleading to have a function named `get` that removes the thing after getting it. Maybe `pop` would be a better name (`popAfterLoginRedirectLogin`). What do you think?

Reply:
Agreed, this would give more context about what it does